ruby/red-arrow/test/test-record-batch.rb

```diff
@@ -178,5 +178,29 @@ def setup
                      @record_batch[[:c, "a", -1, 3..4]])
       end
     end
+
+    sub_test_case("#each_raw_record") do
```

Comment:
I was uncertain about the appropriate location to add the following test cases. 
I added it here because `each_raw_record` is one of the methods for Arrow::Table or Arrow::RecordBatch.
- ruby/red-arrow/test/each-raw-record/*
- ruby/red-arrow/test/test-record-batch.rb

Reply:
We can always use no block version like this:

```diff
diff --git a/ruby/red-arrow/test/each-raw-record/test-basic-arrays.rb b/ruby/red-arrow/test/each-raw-record/test-basic-arrays.rb
index dbbbd79ee..05f0e90ef 100644
--- a/ruby/red-arrow/test/each-raw-record/test-basic-arrays.rb
+++ b/ruby/red-arrow/test/each-raw-record/test-basic-arrays.rb
@@ -22,12 +22,8 @@ module EachRawRecordBasicArraysTests
       [nil],
       [nil],
     ]
-    iterated_records = []
     target = build({column: :null}, records)
-    target.each_raw_record do |record|
-      iterated_records << record
-    end
-    assert_equal(records, iterated_records)
+    assert_equal(records, target.each_raw_record.to_a)
   end
 
   def test_boolean
```

Because a returned `Enumerator` uses with-block version.

Reply:
fix: [bc23b9c](https://github.com/apache/arrow/pull/37600/commits/bc23b9c8e3c91e09b4023b057294663919ff7dac)

I misunderstood that we explicitly had to call the 'each_raw_record' with a block because I didn't understand how Enumerable#to_a worked.

I will unify tests with raw_records and each_raw_record at the following issue.
- https://github.com/apache/arrow/issues/37563